python_modules/dagit/dagit_tests/test_multiprocessing.py

```diff
@@ -0,0 +1,182 @@
+import time
+import gevent
+from dagster import (
+    DependencyDefinition,
+    Field,
+    PipelineContextDefinition,
+    PipelineDefinition,
+    InputDefinition,
+    OutputDefinition,
+    RepositoryDefinition,
+    SolidDefinition,
+    check,
+    lambda_solid,
+    types,
+)
+from dagster.core.execution import (
+    create_execution_plan,
+)
+from dagster.core.evaluator import evaluate_config_value
+from dagster.utils import script_relative_path
+import dagster.pandas as dagster_pd
+
+from dagit.pipeline_execution_manager import MultiprocessingExecutionManager
+from dagit.pipeline_run_storage import PipelineRun, PipelineRunStatus
+
+
+def test_running():
+    run_id = 'run-1'
+    pipeline = define_passing_pipeline()
+    config = evaluate_config_value(
+        pipeline.environment_type, {
+            'solids': {
+                'load_num_csv': {
+                    'config': {
+                        'path': script_relative_path('num.csv'),
+                    }
+                },
+            },
+        }
+    )
+    pipeline_run = PipelineRun(
+        run_id, 'pandas_hello_world', config, create_execution_plan(pipeline, config.value)
+    )
+    execution_manager = MultiprocessingExecutionManager()
+    execution_manager.execute_pipeline(pipeline, config.value, pipeline_run)
+    execution_manager.join()
+    assert pipeline_run.status == PipelineRunStatus.SUCCESS
+    assert len(pipeline_run.all_logs()) > 0
+
+
+def test_failing():
+    run_id = 'run-1'
+    pipeline = define_failing_pipeline()
+    config = evaluate_config_value(
+        pipeline.environment_type, {
+            'solids': {
+                'load_num_csv': {
+                    'config': {
+                        'path': script_relative_path('num.csv'),
+                    }
+                },
+            },
+        }
+    )
+    pipeline_run = PipelineRun(
+        run_id, 'pandas_hello_world', config, create_execution_plan(pipeline, config.value)
+    )
+    execution_manager = MultiprocessingExecutionManager()
+    execution_manager.execute_pipeline(pipeline, config.value, pipeline_run)
+    execution_manager.join()
+    assert pipeline_run.status == PipelineRunStatus.FAILURE
+    assert len(pipeline_run.all_logs()) > 0
+
+
+def test_execution_crash():
+    run_id = 'run-1'
+    pipeline = define_sleepy_pipeline()
+    config = evaluate_config_value(
+        pipeline.environment_type, {
+            'solids': {
+                'load_num_csv': {
+                    'config': {
+                        'path': script_relative_path('num.csv'),
+                    }
+                },
+            },
+        }
+    )
+    pipeline_run = PipelineRun(
+        run_id, 'pandas_hello_world', config, create_execution_plan(pipeline, config.value)
+    )
+    execution_manager = MultiprocessingExecutionManager()
+    execution_manager.execute_pipeline(pipeline, config.value, pipeline_run)
+    gevent.sleep(0.5)
```

Comment:
This is a place where flakyness will eventually occur.

Reply:
eeek there's gotta be a better way to do this